### PR TITLE
[Bugfix:Autograding] Stop subsequent testcase failure on docker errors

### PR DESCRIPTION
**Pull request by emaicus**
At present if an error occurs as the result of a failed docker operation (e.g, container creation), any hanging containers are not cleaned up. This means that subsequent testcases fail due to docker name collisions.

This PR adds container cleanup before testcases are run, intermediately between each testcase, and after execution has finished.

### CODE REVIEW

File: autograder/autograder/execution_environments/container_network.py [hunk 1]
```diff
@@ -793,6 +793,8 @@ def execute_helper(self, containers, script, arguments, logfile):
             self.log_message("ERROR: Could not verify execution mode status.")
             return
 
+        autograding_utils.cleanup_stale_containers(self.untrusted_user)
```
Comment (hdrodz97) [inline autograder/autograder/execution_environments/container_network.py:796]: Is there any reason this isn't inside a `finally` block with the previous `try`...`except`?

**Review by hdrodz97 — COMMENTED** [review]
Overal seems to work well. Only have one minor comment and then I'll feel good about approving this!

**Comment by bmcutler** [comment]
@emaicus 
Status on this PR?

File: autograder/autograder/execution_environments/container_network.py (hunk 1 above)
Comment (emaicus) [reply autograder/autograder/execution_environments/container_network.py]: Moved.

**Pull request merged**